Redesign edit profile page

marlo-longley opened the pull request:
Fixes #447 

The main change of this PR is allowing users to edit non-password items (i.e. name, email) without entering their current password. 

@ggeisler The spacing between form items is slightly taller than your design indicated but close. The helper gem that provides `bootstrap_form_for` is adding extra stuff into the form template that conflicts with CSS I wrote previously for the Devise forms. I personally don't like this gem and would be down to rip out the dependency, but I think refactoring the forms is a separate low-priority issue. I can definitely do a workaround for the spacing however, if you think it's not acceptable.

<img width="756" alt="Screen Shot 2022-03-31 at 15 54 29" src="https://user-images.githubusercontent.com/1328900/161138914-5f9237ab-3cf3-47a2-88c8-88f535c5a8d9.png">

I appreciate your analysis in the issue and agree that for now, a single vertical layout is easiest, rather than implementing tabs. Doing tabs introduced complications with routing.

ggeisler commented:
@marlo-longley Sure, I think the biggest priority was to separate the first three fields from the change password related fields, because that part is just confusing no matter what you are trying to do. So this fixes that.

I'd still like to see a tabbed UI and better spacing (e.g., there appears to be more vertical spacing below the "Change password" heading than the "Cancel account" heading) but I agree that's lower-priority given all of the other things we're trying to update in the application now.

marlo-longley commented:
Thanks @ggeisler - I just fixed the vertical spacing issue for the Cancel button that you caught. Thanks!

corylown commented on spec/features/edit_profile_spec.rb:
```diff
@@ -0,0 +1,28 @@
+      fill_in 'user_name', with: 'Nice Name'
+      click_on 'Update'
+      # successful update redirects to root page
+      expect(page).to have_current_path('/')
```
Maybe this?

```suggestion
      expect(page).to have_content 'Your account has been updated successfully.'
```

corylown commented on spec/features/edit_profile_spec.rb:
```diff
@@ -0,0 +1,28 @@
+      visit '/users/edit'
+      fill_in 'user_password', with: '123'
+      click_on 'Change Password'
+      expect(page).to have_content "Current password can't be blank"
```
Single quotes for consistency:

```suggestion
      expect(page).to have_content 'Current password can't be blank'
```

corylown reviewed (changes requested):
A couple of inline suggestions. Gary's mockups use different language for the new password and password confirmation fields and I'm wondering if we want to make that change in the PR:
<img width="704" alt="Screen Shot 2022-04-01 at 5 04 58 PM" src="https://user-images.githubusercontent.com/458247/161341365-8475683c-d7f8-404b-9f97-4c2f77d6d0d3.png">

marlo-longley commented:
@corylown thanks! The labels for password fields were addressed in a PR by Nick yesterday, so that's why there was a merge conflict. I rebased; should be resolved. I also changed the test as you suggested.

corylown reviewed (changes requested):
@marlo-longley sorry! one more thing I just noticed:

This file `spec/routing/registrations_routing.spec.rb` should be named `spec/routing/registrations_routing_spec.rb`
